tests/vmi_lifecycle_test.go

```diff
@@ -890,17 +871,15 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 			})
 
 			It("[test_id:1638]the vmi with node affinity and anti-pod affinity should not be scheduled", func() {
-				vmi := libvmi.NewCirros()
-				addNodeAffinityToVMI(vmi, nodes.Items[0].Name)
+				vmi := libvmi.NewCirros(libvmi.WithNodeSelectorFor(&nodes.Items[0]))
 				vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
 				Expect(err).ToNot(HaveOccurred(), "Should create VMI")
 				tests.WaitForSuccessfulVMIStart(vmi)
 				curVMI, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred(), "Should get VMI")
 				Expect(curVMI.Status.NodeName).To(Equal(nodes.Items[0].Name), "VMI should run on the same node")
 
-				vmiB := libvmi.NewCirros()
-				addNodeAffinityToVMI(vmiB, nodes.Items[0].Name)
+				vmiB := libvmi.NewCirros(libvmi.WithNodeSelectorFor(&nodes.Items[0]))
 
 				vmiB.Spec.Affinity.PodAntiAffinity = &k8sv1.PodAntiAffinity{
```

Comment:
darn, here `vmiB.Spec.Affinity` is still none. I'll have to rework `libvmi.WithNodeSelectorFor` to set `vmi.Spec.Affinity` instead of old-style `vmi.Spec.NodeSelector`.